nautobot/extras/api/views.py

```diff
@@ -381,3 +387,27 @@ class RelationshipAssociationViewSet(ModelViewSet):
     queryset = RelationshipAssociation.objects.all()
     serializer_class = serializers.RelationshipAssociationSerializer
     filterset_class = filters.RelationshipAssociationFilterSet
+
+
+#
+# GraphQL Queries
+#
+
+
+class GraphQLQueryViewSet(ModelViewSet):
+    queryset = GraphQLQuery.objects.all()
+    serializer_class = serializers.GraphQLQuerySerializer
+    filterset_class = filters.GraphQLQueryFilterSet
+
+    @swagger_auto_schema(method="post", request_body=serializers.GraphQLQuerySerializer)
+    @action(detail=True, methods=["post"])
+    def run(self, request, pk):
+        query_object = get_object_or_404(self.queryset, slug=pk)
+        schema = graphene_settings.SCHEMA
+        backend = get_default_backend()
+        try:
+            document = backend.document_from_string(schema, query_object.query)
+            result = document.execute(context_value=request, variable_values=request.data).to_dict()
+            return Response(result)
+        except GraphQLError as error:
+            return Response({"error": error})
```

Comment:
When there is an error, it would be great if we can return the error message in the same error format as the main GraphQL API
Currently the graphql api will always return a list of errors, stored under the top level key `errors` 
You can see the current implementation here
https://github.com/nautobot/nautobot/blob/develop/nautobot/core/api/views.py#L431